pkg/controller/configmap/configmap_controller.go

```diff
@@ -382,19 +384,20 @@ func (r *ReconcileConfigMap) Reconcile(request reconcile.Request) (reconcile.Res
 		}
 	}
 
+	ncpDeploymentChanged := false
 	if !needChange.ncp && !needChange.agent && !needChange.bootstrap {
 		// Check if NCP_IMAGE or nsx-ncp replicas changed
 		anyMonitoredResDeleted, err := r.isAnyMonitoredNCPResDeleted()
 		if err != nil {
 			return reconcile.Result{Requeue: true}, err
 		} else if !anyMonitoredResDeleted {
-			needChange.ncp, err = r.isNcpDeploymentChanged(ncpReplicas)
+			ncpDeploymentChanged, err = r.isNcpDeploymentChanged(ncpReplicas)
 			if err != nil {
 				return reconcile.Result{Requeue: true}, err
 			}
 
-			if !needChange.ncp {
-				log.Info("no new configuration needs to apply")
+			if !ncpDeploymentChanged {
+				log.Info("no new nsx-ncp deployment needs to apply")
```

Comment:
how about change as ncpDeploymentChanged since this flag is used to detect both ncp image and replicas changes.